### PR TITLE
Update Dilemma 3x5+3 configs.

### PR DESCRIPTION
**Pull request by casuanoob**
#### RGB
- [x] Fix Key Matrix to LED index.
- [x] Fix LED index to physical position.
- [x] Add defines to enable RGB framebuffer animations and other new animations
#### Via Keymap
- [x] Increment USB PID for Via compatibility
- [x] Add encoder configs to `config.h`
- [x] Add `bool encoder_update_kb` function in `3x5_3.c`
- [x] Add `via` keymap (with `encoder_map`)
- [x] Add bootmagic config for right half
- [x] Remove Auto Pointer Layer custom feature

### CODE REVIEW

File: keyboards/bastardkb/dilemma/3x5_3/keymaps/via/config.h [hunk 1]
```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright 2021 Charly Delay <charly@codesink.dev> (@0xcharly)
```
Comment (0xcharly) [inline keyboards/bastardkb/dilemma/3x5_3/keymaps/via/config.h:2]: You should update those to put your name in them too!

File: keyboards/bastardkb/dilemma/3x5_3/keymaps/via/config.h [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ */
+#define IGNORE_MOD_TAP_INTERRUPT
+
+/* Charybdis-specific features. */
+
+#ifdef POINTING_DEVICE_ENABLE
+// Automatically enable the pointer layer when moving the trackpad.
+//#define POINTING_DEVICE_AUTO_MOUSE_ENABLE
+//#define AUTO_MOUSE_DEFAULT_LAYER 3
+#endif // POINTING_DEVICE_ENABLE
```
Comment (0xcharly) [inline keyboards/bastardkb/dilemma/3x5_3/keymaps/via/config.h:42]: Can we remove these?

If they are QMK features, they should be well documented, so no need to leave a trail here.

File: keyboards/bastardkb/dilemma/3x5_3/rules.mk [hunk 3]
```diff
@@ -6,10 +6,14 @@ MOUSEKEY_ENABLE = yes       # Mouse keys
 EXTRAKEY_ENABLE = yes       # Audio control and System control
 CONSOLE_ENABLE = no         # Console for debug
 COMMAND_ENABLE = no         # Commands for debug and configuration
-NKRO_ENABLE = no            # Enable N-Key Rollover
+NKRO_ENABLE = yes           # Enable N-Key Rollover
```
Comment (0xcharly) [inline keyboards/bastardkb/dilemma/3x5_3/rules.mk:9]: Is this intentional or was this for tests?

File: keyboards/bastardkb/dilemma/3x5_3/rules.mk [hunk 4]
```diff
@@ -6,10 +6,14 @@ MOUSEKEY_ENABLE = yes       # Mouse keys
...
 BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
 RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
 AUDIO_ENABLE = no           # Audio output
+MAGIC_ENABLE = yes			# Magic keycodes
+TRI_LAYER_ENABLE = yes		# Tri Layers
+CAPS_WORD_ENABLE = yes		# Caps Word
```
Comment (0xcharly) [inline keyboards/bastardkb/dilemma/3x5_3/rules.mk:15]: Should these be enabled by default if they're not used in the stock keymaps?

File: keyboards/bastardkb/dilemma/3x5_3/rules.mk [hunk 5]
```diff
@@ -25,5 +29,7 @@ POINTING_DEVICE_DRIVER = cirque_pinnacle_spi
 
 SPLIT_KEYBOARD = yes
 
+DEBOUNCE_TYPE = asym_eager_defer_pk
+
```
Comment (0xcharly) [inline keyboards/bastardkb/dilemma/3x5_3/rules.mk:33]: I'm wondering if we should leave this sort of optimization to the user's discretion.

The rational being that the stock keymaps should showcase the keyboard's feature, but use as much of the defaults as possible.

What do you think?

File: keyboards/bastardkb/dilemma/3x5_3/keymaps/via/readme.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+```c
+#define DILEMMA_AUTO_POINTER_LAYER_TRIGGER_TIMEOUT_MS 1000
+```
+
+The trigger sensibility can also be tuned. The lower the value, the more sensible the trigger:
+
+```c
+#define DILEMMA_AUTO_POINTER_LAYER_TRIGGER_THRESHOLD 8
+```
```
Comment (0xcharly) [inline keyboards/bastardkb/dilemma/3x5_3/keymaps/via/readme.md:67]: Can we remove these now that we're using QMK's implementation?

**Review by 0xcharly — COMMENTED** [review]
Thanks, @casuanoob! Looking good!

File: keyboards/bastardkb/dilemma/3x5_3/rules.mk (hunk 3 above)
Comment (casuanoob) [reply keyboards/bastardkb/dilemma/3x5_3/rules.mk:9]: Intentional, `NKRO_ENABLE = yes` will still default to NKRO off, but in combination with `MAGIC_ENABLE = yes` will allow Via-only users to have the option of toggling NKRO on if they wish, by assigning the magic keycodes.

File: keyboards/bastardkb/dilemma/3x5_3/keymaps/via/config.h (hunk 2 above)
Comment (casuanoob) [reply keyboards/bastardkb/dilemma/3x5_3/keymaps/via/config.h:42]: yep, will remove

File: keyboards/bastardkb/dilemma/3x5_3/keymaps/via/readme.md (hunk 6 above)
Comment (casuanoob) [reply keyboards/bastardkb/dilemma/3x5_3/keymaps/via/readme.md:67]: yep, I completely forgot to change the readme at first

**Pull request merged**